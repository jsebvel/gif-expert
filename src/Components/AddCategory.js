import { useState } from "react";
import Proptypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2){
            setCategories(lastCategories => [ inputValue, ...lastCategories]);
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <input type="text"
                value={inputValue}
                onChange={(e) => {
                    handleInputChange(e);
                }}
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: Proptypes.func.isRequired
}