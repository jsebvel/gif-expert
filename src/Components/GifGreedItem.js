import PropTypes from 'prop-types';
export const GifGreedItem = ({ title, url }) => {
    return (
        <div className="card ">
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    )
}

GifGreedItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
    