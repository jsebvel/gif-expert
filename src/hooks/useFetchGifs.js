import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category)
            .then(data => {
                setTimeout(() => {
                    console.log(data);
                    setState({
                        data: data,
                        loading: false
                    });

                }, 3000);
            });
    }, [])


    return state;
}