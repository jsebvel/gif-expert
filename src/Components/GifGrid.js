import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGreedItem } from "./GifGreedItem";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([])
    // useEffect(() => {
    //     getGifs(category).then(imgs => setImages(imgs));
    // }, [category])
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading && <p className="animate_animated animate_fadeIn">{category}</p>}
            <div className="card-grid animate__animated animate__bounce">
                {
                    images.map(img =>
                        <GifGreedItem
                            key={img.id}
                            {...img} />
                    )
                }
            </div>
        </>
    )
}

GifGrid.protoTypes = {
    category: PropTypes.string.isRequired
}
