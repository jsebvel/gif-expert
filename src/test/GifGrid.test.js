import { shallow } from 'enzyme';
import { GifGrid } from '../Components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../hooks/useFetchGifs');


describe('Test on GifGrid', () => {
    const category = 'guitar';
    // beforeEach(() => {
    //     wrapper = shallow(<GifGrid category={'anyCat'}></GifGrid>)
    // })
    test('Snapshot should match', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show items where images are loaded', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false,
        });

        const gifs = [
            {
                id: 'ABC',
                url: 'https://anything',
                title: 'any title'

            },
            {
                id: 'ABCD',
                url: 'https://anythingD',
                title: 'any titleD'

            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs, 
            loading: false
        })
        const wrapper = shallow(<GifGrid category= {category} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGreedItem').exists()).toBe(true);
        expect(wrapper.find('GifGreedItem').length).toBe(gifs.length);
    });
});
