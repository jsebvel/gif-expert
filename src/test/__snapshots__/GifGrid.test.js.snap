// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid Should show items where images are loaded 1`] = `
<Fragment>
  <h3>
    guitar
  </h3>
  <div
    className="card-grid animate__animated animate__bounce"
  >
    <GifGreedItem
      id="ABC"
      key="ABC"
      title="any title"
      url="https://anything"
    />
    <GifGreedItem
      id="ABCD"
      key="ABCD"
      title="any titleD"
      url="https://anythingD"
    />
  </div>
</Fragment>
`;

exports[`Test on GifGrid Snapshot should match 1`] = `
<Fragment>
  <h3>
    guitar
  </h3>
  <p
    className="animate_animated animate_fadeIn"
  >
    guitar
  </p>
  <div
    className="card-grid animate__animated animate__bounce"
  />
</Fragment>
`;
